pkg/types/azure/validation/machinepool.go

```diff
@@ -221,6 +235,91 @@ func validateOSImage(p *azure.MachinePool, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// getDataDisksSecurityProfileType extracts the security encryption type from data disks
+// to ensure consistency with OS disk security settings.
+//
+// Security profile requirements (from CAPZ types.go):
+// - When set to VMGuestStateOnly: VirtualizedTrustedPlatformModule should be Enabled
+// - When set to DiskWithVMGuestState: EncryptionAtHost should be disabled,
+//   SecureBoot and VirtualizedTrustedPlatformModule should be Enabled
+// - DiskWithVMGuestState can only be used with Confidential VMs
+//
+// Note: All data disks should have the same security encryption type to avoid
+// configuration conflicts and ensure consistent security posture.
```

Comment:
:question: This comment says `All data disks should have the same security encryption type`, but there is no validation for it, right? Should we add it?